woocommerce/woocommerce
# Added logic to draft and publish the ready page on each run

### Changes proposed in this Pull Request:

This PR proposes using the WordPress API to change the Ready page's status from `Published` to `Draft` at the start of the test run to prevent concurrent tests from running. Once the tests are done, the page's status is switched back to `Published` for subsequent test runs.

### How to test the changes in this Pull Request:

1. Make sure the CI passes.
2. Run the e2e tests locally once. While the tests are running verify the Ready page's status is in `Draft` and not `Publish`. Verify the tests pass.
3. Without making any changes, run the tests again (optionally use `E2E_RETEST=1 npx wc-e2e test:e2e` when doing so; though there's no need to run the entire suite) and verify the tests run and don't give logs that the Docker container is still being built.

**Test blocking of concurrent runs:**

1. Prepare two terminal windows
2. In the first terminal window, start the entire e2e test suite with `npx wc-e2e test:e2e`
3. In the second terminal window, while the e2e tests are running, start another run of the e2e tests with `npx wc-e2e test:e2e`
4. In the first window, verify the tests are still running
5. In the second window, verify you see `Docker container is still being built` periodically being output to the console

https://user-images.githubusercontent.com/71906536/122657529-80cd5b80-d121-11eb-9984-b0f5f6de2964.mov

### Changelog entry

N/A
